### PR TITLE
Activity Infrastructure: Initial `USNat` support

### PR DESCRIPTION
**Pull request by CTMBNara**

### CODE REVIEW

File: src/main/java/org/prebid/server/activity/infrastructure/payload/impl/BidRequestActivityCallPayload.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public class BidRequestActivityCallPayload implements GeoActivityCallPayload {
+
+    @Delegate
+    ActivityCallPayload componentInfo;
+
+    BidRequest bidRequest;
+
+    @Override
+    public String country() {
+        return Optional.ofNullable(bidRequest)
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/activity/infrastructure/payload/impl/BidRequestActivityCallPayload.java:25]: Is it possible taht bidRequest will be null in some way?

File: src/main/java/org/prebid/server/auction/gpp/AuctionGppService.java [hunk 2]
```diff
@@ -23,37 +26,74 @@ public AuctionGppService(GppService gppService) {
...
+        return regsOptional
+                .map(Regs::getGdpr)
+                .or(() -> regsOptional
+                        .map(Regs::getExt)
+                        .map(ExtRegs::getGdpr))
+                .orElse(null);
+    }
+
+    private static String consent(User user) {
+        final Optional<User> userOptional = Optional.ofNullable(user);
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/auction/gpp/AuctionGppService.java:67]: Looks like this variable is redundant

File: src/main/java/org/prebid/server/auction/gpp/AuctionGppService.java [hunk 3]
```diff
@@ -23,37 +26,74 @@ public AuctionGppService(GppService gppService) {
...
+        return userOptional
+                .map(User::getConsent)
+                .or(() -> userOptional
+                        .map(User::getExt)
+                        .map(ExtUser::getConsent))
+                .orElse(null);
+    }
+
+    private static String usPrivacy(Regs regs) {
+        final Optional<Regs> regsOptional = Optional.ofNullable(regs);
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/auction/gpp/AuctionGppService.java:78]: Looks like this variable is redundant

File: src/main/java/org/prebid/server/auction/gpp/SetuidGppService.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        final UpdateResult<Integer> updatedGdpr = updateResult(
+                toInt(privacy.getGdpr()),
+                tcfEuV2Privacy != null ? tcfEuV2Privacy.getGdpr() : null);
+        final UpdateResult<String> updatedConsent = updateResult(
+                privacy.getConsentString(),
+                tcfEuV2Privacy != null ? tcfEuV2Privacy.getConsent() : null);
+
+        return updatedGdpr.isUpdated() || updatedConsent.isUpdated()
+
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/auction/gpp/SetuidGppService.java:69]: Redundant \n

File: src/main/java/org/prebid/server/spring/config/ActivityInfrastructureConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Bean
+    ActivityRuleFactory activityRuleFactory(List<RuleCreator<?>> ruleCreators) {
+        return new ActivityRuleFactory(ruleCreators);
+    }
+
+    @Bean
+    ActivityInfrastructureCreator activityInfrastructureCreator(ActivityRuleFactory activityRuleFactory,
+                                                                Metrics metrics) {
+
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/spring/config/ActivityInfrastructureConfiguration.java:35]: Redundant \n

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/auction/gpp/AuctionGppService.java (hunk 2 above)
Comment (CTMBNara) [reply src/main/java/org/prebid/server/auction/gpp/AuctionGppService.java:67]: Why do you think so?

File: src/main/java/org/prebid/server/auction/gpp/AuctionGppService.java (hunk 3 above)
Comment (CTMBNara) [reply src/main/java/org/prebid/server/auction/gpp/AuctionGppService.java:78]: Why do you think so?

File: src/main/java/org/prebid/server/activity/infrastructure/creator/ActivityInfrastructureCreator.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+    public ActivityInfrastructure create(Account account, GppContext gppContext, TraceLevel traceLevel) {
+        return new ActivityInfrastructure(
+                account.getId(),
+                parse(account, gppContext),
+                traceLevel,
+                metrics);
+    }
+
+    Map<Activity, ActivityConfiguration> parse(Account account, GppContext gppContext) {
```
[the remark this answers is not in the document]
Comment (CTMBNara) [reply src/main/java/org/prebid/server/activity/infrastructure/creator/ActivityInfrastructureCreator.java:44]: The default access modifier has been used here to be able to use this method in unit tests.
This approach is necessary to check that all activities are present in the returned map.

File: src/main/java/org/prebid/server/spring/config/ActivityInfrastructureConfiguration.java (hunk 5 above)
Comment (CTMBNara) [reply src/main/java/org/prebid/server/spring/config/ActivityInfrastructureConfiguration.java:35]: We usually add \n after any multi-line declaration of a method.

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**